blazor/datagrid/connecting-to-data-source/microsoft-sql-server.md

```diff
@@ -0,0 +1,641 @@
+
+**Delete Operation:**
+
+To perform the delete operation, you need to override the `Remove` or `RemoveAsync` method of the custom adaptor. Below is the code snippet that you can add to the **CustomAdaptorComponent.razor** file. This code snippet demonstrates how to handle the deletion of existing records within the `RemoveAsync` method of the custom adaptor component. Adjust the logic inside this method according to your application requirements.
+
+{% tabs %}
+{% highlight razor tabtitle="Index.razor"%}
+
+public override async Task<object> RemoveAsync(DataManager dm, object Value, string keyField, string key)
+{
+    // Add your delete logic here
+    // This method will be invoked when deleting existing records from the grid
+    await Order.RemoveOrderAsync(Value as int?);
+    return Value;
+}
+
+{% endhighlight %}
+{% highlight razor tabtitle="Orderdata.cs"%}
+public async Task RemoveOrderAsync(int? Value)
+{
+    string connectionString = @"<Enter a valid connection string>"; 
+    string QueryStr = $"Delete from Orders where OrderID={Value}";
+    SqlConnection Con = new SqlConnection(connectionString);
+    Con.Open();
+    SqlCommand Cmd = new SqlCommand(QueryStr, Con);
+    Cmd.ExecuteNonQuery();
+    Con.Close();
+}
+{% endhighlight %}
+{% endtabs %}
+
+> You can find the sample in this [GitHub location](https://github.com/VigneshNatarajan27/connecting-microsoft-sql-data-base-to-data-grid).
+
+## Binding data from Microsoft SQL Server using an API service
+
+### Creating an API service 
+
+**1.** Open Visual Studio and create an ASP.NET Core Web App project type, naming it **MyWebService**. To create an ASP.NET Core Web application, follow the documentation [link](https://learn.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-aspnet-core?view=vs-2022).
+
+![Create ASP.NET Core Web App project](../images/azure-asp-core-web-service-create.png)
+
+**2.** To connect a Microsoft SQL using the **System.Data.SqlClient** in our application, we need to install the [System.Data.SqlClient](https://www.nuget.org/packages/System.Data.SqlClient/4.8.6?_src=template) NuGet package. To do so, open the NuGet package manager of the project solution, search for the package **System.Data.SqlClient** and install it.
+
+![Add the NuGet package Sytem.Data.SqlClient to the project](../images/system-Data-sql-client-nuget-package-install.png)
+
+**3.** Create a API controller (aka, GridController.cs) file under **Controllers** folder that helps to establish data communication with the DataGrid.
+
+**4.** In the API controller  (aka, GridController), connect to Microsoft SQL server. In the **Get()** method **SqlConnection** helps to connect the SQL database (that is, Database1.mdf). Next, using **SqlCommand** and **SqlDataAdapter** you can process the desired SQL query string and retrieve data from the database. The **Fill** method of the DataAdapter is used to populate the SQL data into a **DataTable** as shown in the following code snippet.
```

Comment:
Need to provide the datamanger related package details to access our dataoperations class methods

Reply:
Resolved